pkg/logger/logger.go

```diff
@@ -0,0 +1,289 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package logger
+
+import (
+	"os"
+	"sync"
+	"time"
+
+	"github.com/dapr/dapr/pkg/version"
+	"github.com/dapr/dapr/utils"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	// LogTypeLog is normal log type
+	LogTypeLog = "log"
+	// LogTypeRequest is Request log type
+	LogTypeRequest = "request"
+
+	// Field names that defines Dapr log schema
+	logFieldTimeStamp = "time"
+	logFieldLevel     = "level"
+	logFieldType      = "log_type"
+	logFieldScope     = "scope"
+	logFieldMessage   = "message"
+	logFieldInstance  = "instance"
+	logFieldDaprVer   = "dapr_ver"
+	logFieldDaprID    = "dapr_id"
+)
+
+// LogLevel is Dapr Logger Level type
+type LogLevel int
+
+const (
+	// DebugLevel has verbose message
+	DebugLevel LogLevel = iota
+	// InfoLevel is default log level
+	InfoLevel
+	// WarnLevel is for logging messages about possible issues
+	WarnLevel
+	// ErrorLevel is for logging errors
+	ErrorLevel
+	// FatalLevel is for logging fatal messages. The system shutsdown after logging the message.
+	FatalLevel
+)
+
+var daprLoggingLevelToLogrusLevel = map[LogLevel]logrus.Level{
```

Comment:
Why can't we just expose `logrus.Level` instead of creating a mapping layer?

Reply:
We can, but having `LogLevel`, the other source codes (except for logger.go) do not need to import any logrus pkgs. It makes sense when we create the helper.

Maybe we can do type-cast from `logrus.Level` to `LogLevel` because both are same uint32. But I normally prefer mapping to casting between two different types in this case.

Reply:
See my new comments about this.